src/main/java/com/commercetools/sync/producttypes/helpers/BatchProcessor.java

```diff
@@ -0,0 +1,154 @@
+package com.commercetools.sync.producttypes.helpers;
+
+import com.commercetools.sync.commons.exceptions.InvalidProductTypeDraftException;
+import com.commercetools.sync.commons.exceptions.InvalidReferenceException;
+import com.commercetools.sync.producttypes.ProductTypeSync;
+import io.sphere.sdk.products.attributes.AttributeDefinitionDraft;
+import io.sphere.sdk.products.attributes.AttributeType;
+import io.sphere.sdk.products.attributes.NestedAttributeType;
+import io.sphere.sdk.products.attributes.SetAttributeType;
+import io.sphere.sdk.producttypes.ProductTypeDraft;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.commercetools.sync.commons.helpers.BaseReferenceResolver.BLANK_ID_VALUE_ON_RESOURCE_IDENTIFIER;
+import static java.lang.String.format;
+import static java.util.Collections.emptySet;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class BatchProcessor {
```

Comment:
that is true and good catch! 

IMO this is better than the other class names like `ProductTypeReferenceResolver` since it is already under the `producttypes` package so actually `ReferenceResolver` would be enough if it is under this package to know which reference resolver it is. So the prefix is redundant and making the class name longer for no reason. 

`BatchProcessor` for products as well has the name like that. So to be consistent, the productType one also is the same.

But to be consistent throughout, I changed both: f2d28bb

Although I also don't like that it is `ProductTypeBatchProcessor` not `ProductTypeDraftBatchProcessor`.. But oh well, too late for this..

Reply:
I agree with you the naming from the package is enough but yes it's better to be consistent. btw could it be a breaking change for the `ProductBatchProcessor`?

Reply:
no as this class is never documented and not in release notes. Because this class is only meant for internal use not external yes.